Source/Scene/Model.js

```diff
@@ -4322,6 +4376,84 @@ define([
         }
     };
 
+    function destroyIfNotCached(rendererResources, cachedRendererResources) {
+        if (rendererResources.programs !== cachedRendererResources.programs) {
+            destroy(rendererResources.programs);
+        }
+        if (rendererResources.pickPrograms !== cachedRendererResources.pickPrograms) {
+            destroy(rendererResources.pickPrograms);
+        }
+        if (rendererResources.silhouettePrograms !== cachedRendererResources.silhouettePrograms) {
+            destroy(rendererResources.silhouettePrograms);
+        }
+    }
+
+    // Run from update iff:
+    // - everything is loaded
+    // - clipping planes state change (shouldRegenerateShaders) OR color state set
+    // Run this from destructor after removing color state and clipping plane state
+    function regenerateShaders(model, frameState) {
+        // In regards to _cachedRendererResources:
+        // Fair to assume that this is data that should just never get modified due to clipping planes or model color.
+        // So if clipping planes or model color active:
+        // - delink _rendererResources.*programs and create new dictionaries.
+        // - do NOT destroy any programs - might be used by copies of the model or by might be needed in the future if clipping planes/model color is deactivated
+
+        // If clipping planes/model color inactive:
+        // - destroy _rendererResources.*programs
+        // - relink _rendererResources.*programs to _cachedRendererResources
```

Comment:
If a model has clipping planes and then a minute later has its color set, what happens to the clipping planes-only shader?

Reply:
Ah... I guess this should always be running `destroyIfNotCached`